generator/src/main/java/com/scottlogic/deg/generator/generation/StandardFieldValueSourceEvaluator.java

```diff
@@ -108,20 +108,42 @@ private FieldValueSource getNumericSource(FieldSpec fieldSpec) {
             ? new NumericRestrictions()
             : fieldSpec.getNumericRestrictions();
 
-        int numericScale = fieldSpec.getGranularityRestrictions() != null
-            ? fieldSpec.getGranularityRestrictions().getNumericScale()
-            : 0;
+        int numericScale = getNumericScale(restrictions, fieldSpec);
 
-        if (numericScale == 0) {
+        if ((restrictions.min == null || restrictions.max == null) || isIntegerFieldValue(restrictions, numericScale)) {
             return new IntegerFieldValueSource(
                 restrictions,
                 getBlacklist(fieldSpec));
-        } else {
-            return new RealNumberFieldValueSource(
-                restrictions,
-                getBlacklist(fieldSpec),
-                numericScale);
         }
+
+        return new RealNumberFieldValueSource(
+            restrictions,
+            getBlacklist(fieldSpec),
+            numericScale);
+    }
+
+    private int getNumericScale(NumericRestrictions numericRestrictions, FieldSpec fieldSpec) {
```

Comment:
this would probably be better either
only passing fieldSpec
or
passing numericRestrictions, and granularityRestrictions